account_payment_term_partner_holiday/models/account_payment_term.py

```diff
@@ -2,24 +2,43 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
 from odoo import api, fields, models
+from dateutil.relativedelta import relativedelta
 
 
 class AccountPaymentTerm(models.Model):
     _inherit = "account.payment.term"
 
     @api.one
     def compute(self, value, date_ref=False):
+        """If date_to_check has changed is because initial date is in holiday
+        period and these change make necessary to re-compute again to be sure
+        that date returnes is according to payment term (+10 days or custom
+        payment days).
+        """
```

Comment:
As with before, maybe a more complete explanation would be clearer here:
```suggestion
        """Compute the due date taking into account the holiday periods
        set in the partner.
         
        Once an initial date resulting of the payment term is computed, compute the first 
        available date after that.
        Then, recompute the due date applying the payment term to the 
        previously computed date.
        """
```